internal/system/agent/init.go

```diff
@@ -74,45 +97,149 @@ func Retry(useRegistry bool, useProfile string, timeout int, wait *sync.WaitGrou
 	return
 }
 
-func Init() bool {
+func Init(useRegistry bool) bool {
 	if Configuration == nil {
 		return false
 	}
+
+	if useRegistry && registryClient != nil {
+		chErrors = make(chan error)
+		chUpdates = make(chan interface{})
+		go listenForConfigChanges()
+	}
+
 	return true
 }
 
-func initializeConfiguration(useProfile string) (*interfaces.ConfigurationStruct, error) {
-	//We currently have to load configuration from filesystem first in order to obtain RegistryHost/Port
-	err := config.LoadFromFile(useProfile, Conf)
+func Destruct() {
+
+	if chErrors != nil {
+		close(chErrors)
+	}
+
+	if chUpdates != nil {
+		close(chUpdates)
+	}
+}
+
+func initializeConfiguration(useRegistry bool, useProfile string) (*ConfigurationStruct, error) {
+	//We currently have to load configuration from filesystem first in order to obtain Registry Host/Port
+	configuration := &ConfigurationStruct{}
+	err := config.LoadFromFile(useProfile, configuration)
 	if err != nil {
 		return nil, err
 	}
 
-	return Conf, nil
+	// Even though we don't conceive of anyone looking up the info for the SMA itself (via the
+	// Registry), we are setting this (Registry) machinery up for the SMA itself to start passing through
+	// some requests to the Registry (for other device and core services).
```

Comment:
In hindsight, that comment became obsolete (since the time I started teasing apart the logic of the implementation) and is outright misleading at this time! Good catch: This one bites the dust 👍